skimage/feature/_cascade.pyx

```diff
@@ -734,7 +734,6 @@ cdef class Cascade:
 
                     if result:
 
-                        new_detection = Detection()
```

Comment:
Does this mean that the declaration above instantiates it?

Reply:
I am not sure what it does (I didn't find documentation calling the struct like this in a quick search). I guess taking a look at the generated C could would be needed to see what it actually does

Reply:
I am thinking about reverting this change just to be safe. This occurs within a parallel range (`prange`) loop and the old notation resulted in another temporary struct being marked as thread private in the OpenMP loop.

I did not realize that there is only a single test case using this function, so I am a bit wary that this may break things!

Reply:
When you assign to a variable within a `prange` loop, Cython will automatically mark the variable as private.

Reply:
I have reverted the change to be more confident that multiple threads will not be assigning to a common struct, creating thread conflicts.